### PR TITLE
[AIRFLOW-3018] Fix Minor issues in Documentation

### PR DESCRIPTION
**Pull request by kaxil**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. 
  - https://issues.apache.org/jira/browse/AIRFLOW-3018

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Documentation contains lots of hooks and operators with mis-spelled, missing types and wrongly formatted text

Few issues:
screen shot 2018-09-05 at 22 14 19
screen shot 2018-09-05 at 22 51 49

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by r39132** [comment]
@kaxil How did you discover and fix most of these? Are you using some sort of auto-linter?

**Comment by r39132** [comment]
@kaxil A google cloud helper bucket test is failing..

**Comment by kaxil** [comment]
@r39132 Unfortunately it was a long manual process :( .  I was scrolling pages (took couple of hours) at https://airflow.apache.org/code.html and https://airflow.apache.org/integration.html to see if everything is formatted properly or something is missed. 

Our documentation still needs some love.

**Comment by kaxil** [comment]
@r39132 I have also updated the PR, so should fix Travis :)

File: airflow/contrib/operators/dataflow_operator.py [hunk 1]
```diff
@@ -149,6 +148,25 @@ class DataflowTemplateOperator(BaseOperator):
...
     will be passed to the job.
+
+    :param template: The reference to the DataFlow template.
+    :type template: string
+    :param dataflow_default_options: Map of default job environment options.
+    :type dataflow_default_options: dict
+    :param parameters: Map of job specific parameters for the template.
+    :type parameters: dict
+    :param gcp_conn_id: The connection ID to use connecting to Google Cloud
+    Platform.
```
Comment (Fokko) [inline airflow/contrib/operators/dataflow_operator.py]: How do we indent? In this docstring there are two ways?

File: airflow/contrib/operators/dataflow_operator.py [hunk 2]
```diff
@@ -33,6 +33,39 @@ class DataFlowJavaOperator(BaseOperator):
...
+        https://cloud.google.com/dataflow/pipelines/specifying-exec-params
+
+    :param jar: The reference to a self executing DataFlow jar.
+    :type jar: string
+    :param dataflow_default_options: Map of default job options.
+    :type dataflow_default_options: dict
+    :param options: Map of job specific options.
+    :type options: dict
+    :param gcp_conn_id: The connection ID to use connecting to Google Cloud
+    Platform.
```
Comment (Fokko) [inline airflow/contrib/operators/dataflow_operator.py]: Indentation :)

File: airflow/contrib/operators/bigquery_to_gcs.py [hunk 3]
```diff
@@ -43,7 +43,7 @@ class BigQueryToCloudStorageOperator(BaseOperator):
     :param compression: Type of compression to use.
     :type compression: string
     :param export_format: File format to export.
-    :type field_delimiter: string
+    :type export_format: string
     :param field_delimiter: The delimiter to use when extracting to a CSV.
     :type field_delimiter: string
```
Comment (Fokko) [inline airflow/contrib/operators/bigquery_to_gcs.py:48]: string -> str?

File: airflow/contrib/operators/bigquery_operator.py [hunk 4]
```diff
@@ -383,7 +383,7 @@ class BigQueryCreateExternalTableOperator(BaseOperator):
     :type schema_fields: list
     :param schema_object: If set, a GCS object path pointing to a .json file that
         contains the schema for the table. (templated)
-    :param schema_object: string
+    :type schema_object: string
```
Comment (Fokko) [inline airflow/contrib/operators/bigquery_operator.py:386]: Please convert this to str as well :)

**Review by Fokko — CHANGES_REQUESTED** [review]
Minor nits :) Thanks for doing this! 👍

File: airflow/contrib/operators/bigquery_operator.py (hunk 4 above)
Comment (kaxil) [reply airflow/contrib/operators/bigquery_operator.py:386]: I am creating a follow up PR to do this as this (`string` -> `str`) is going to affect 100s of files.

File: airflow/contrib/operators/bigquery_to_gcs.py (hunk 3 above)
Comment (kaxil) [reply airflow/contrib/operators/bigquery_to_gcs.py:48]: Same comment: I am creating a follow-up PR to do this as this (`string` -> `str`) is going to affect 100s of files. :)

**Comment by r39132** [comment]
With the exception of the `striing-->str` conversion that will be handled in a separate PR, this looks good to go.

File: airflow/contrib/operators/bigquery_operator.py (hunk 4 above)
Comment (Fokko) [reply airflow/contrib/operators/bigquery_operator.py:386]: Awesome!

**Comment by Fokko** [comment]
There should be a docstring checker, like flake8 for the code style 🤔

**Pull request merged**